common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AndroidTaskStateGenerator.java

```diff
@@ -0,0 +1,43 @@
+package com.microsoft.identity.common.internal.providers.oauth2;
+
+import com.microsoft.identity.common.logging.Logger;
+
+import lombok.Getter;
+import lombok.NonNull;
+
+/**
+ * Encodes the Android Task ID (taskId) into the state sent as part of the authorization request
+ * This allows the response to be correlated to the request.
+ */
+public class AndroidTaskStateGenerator extends DefaultStateGenerator {
+
+    private static final String SPLITTER = ":";
+    private static final String TAG = "AndroidTaskStateGenerator";
+
+    @Getter
+    private int taskId;
+
+    public AndroidTaskStateGenerator(@NonNull final int taskId){
+        this.taskId = taskId;
+    }
+
+    @Override
+    public String generate() {
+        String state = super.generate();
+        state = String.format("%d%s%s", this.taskId, SPLITTER, state);
+        return state;
+    }
+
+    public static int getTaskFromState(String state){
```

Comment:
Looks like this can be an instance method

Reply:
It's more painful for it to be an instance method from a usage point of view.